src/services/base/project-setting.ts

```diff
@@ -53,9 +53,11 @@ export function createAttribute(projectId: string, name: string, dataType: strin
     jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase({ name, dataType })));
 }
 
-export function updateAttribute(projectId: string, attributeId: string, onResult: (result: any) => void, dataType?: string, isPrimaryKey?: boolean, name?: string, sourceCode?: string, visibility?: string) {
+export function updateAttribute(projectId: string, attributeId: string, onResult: (result: any) => void, dataType?: string, isPrimaryKey?: boolean, name?: string, sourceCode?: string, visibility?: string, additionalConfig?: any) {
     const finalUrl = `${projectSettingEndpoint}/${projectId}/update-attribute`;
-    jsonFetchWrapper(finalUrl, FetchType.POST, onResult, JSON.stringify(convertCamelToSnakeCase({ attributeId, dataType, isPrimaryKey, name, sourceCode, visibility })));
+    const baseData: any = convertCamelToSnakeCase({ attributeId, dataType, isPrimaryKey, name, sourceCode, visibility });
+    baseData.additional_config = additionalConfig;
```

Comment:
why is additional_config not added as part of the convert method?

Reply:
the function works recursivly but the values of additional_config need to stay snake case